base/show.jl

```diff
@@ -595,7 +595,11 @@ function show(io::IO, p::Pair)
     for i = (1, 2)
         io_i = IOContext(iocompact, :typeinfo => typeinfos[i])
         isdelimited(io_i, p[i]) || print(io, "(")
-        show(io_i, p[i])
+        if i == 1 && isoperator(Symbol(p[i]))
```

Comment:
Maybe also guard with `(p[i] isa Symbol || p[i] isa Function)`? This will also parenthesize strings, for example, which isn't necessary.

Reply:
Normally this is the job for `isdelimited`: `isdelimited(io, x)` tells whether `x` will be printed with parenthesis around (or brackets, whatever) so that `show` for `Pair` doesn't need to add them itself. So in this case `isdelimited(io, x::Function) = !isoperator(Symbol(x))` should do the trick. I understand that you wanted to discriminate on whether we print the first or 2nd element of the pair, but this doesn't seem important enough for me to bypass the `isdelimited` functionality and hardcode the thing in `show(::Pair)`.

Reply:
Hi @rfourquet. I feel the implementation used in this PR is equivalent to overloading `isdelimited` for `p[1]`. The job of `isdelimited` is to call  `show_default`. Even if it is overloaded, we anyways need to write the if-else code.

Reply:
Thanks for pointing it out @JeffBezanson. I also added a test for checking the display of strings.